enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,274 @@
+
+```yaml
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    monitoring.openshift.io/scrape-profile: full
+  name: foo-full
+  namespace: openshift-bar
+spec:
+  endpoints:
+    port: metrics
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: foo
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    monitoring.openshift.io/scrape-profile: minimal
+  name: foo-minimal
+  namespace: openshift-bar
+spec:
+  endpoints:
+    port: metrics
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: foo
+  metricRelabelings:
+  - sourceLabels: [__name__]
+    action: keep
+    regex: "requests_total|requests_failed_total"
+ ```
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+  // Defines the scraping profile that will be enforced on the platform
+  // prometheus instance. Possible values are full and minimal.
+  //
+  // +kubebuilder:validation:Enum=full;minimal
+  // +optional
+	ScrapeProfile string `json:"scrapeProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Each OpenShift team that wants to adopt this feature will be responsible for providing the different monitors. This work is not trivial. Dependencies between operators and their metrics exist. This makes it difficult for developers to determine whether a given metric must be added to the "minimal" profile or not. To aid teams with this problem we will provide a tool that would consume a monitor and generate the "minimal" monitor for that operator. However for this, the tool would need to have access to an up to date, installation of OpenShift in order to query the Prometheus instance running in the cluster.
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't being used in an alert is now required, how does the monitor responsible for that metric gets updated?
+  - The tool we provide would run in CI and ensure are ServiceMonitors are up to date;
+
+- A new profile is added, what will happen to operators that had implemented scrape profiles but did not implement the latest profile.
+  - TBD. Currently, according to the description above, they would not be scraped when the new profile would be picked.
+
+### Drawbacks
+
+- Extra CI cycles
+
+## Design Details
+
+### Open Questions [optional]
+
+- Should we add future profiles? 
+- Should developers have to comply with all profiles?
+
+### Test Plan
+
+- Unit tests in CMO to validate that the correct monitors are being selected
+- E2E tests in CMO to validate that everything works correctly
+Unsure on this one... (- Testing in openshift/CI to validate that every metrics being used exists in the cluster?)
```

Comment:
I think that we need to explore this space. We already had the idea to detect alerting rules using metrics that don't exist (see https://github.com/openshift/origin/pull/26787).